features/text/0001-multiple_service_endpoints.md

```diff
@@ -0,0 +1,323 @@
+      <LOCAL_ID>2</LOCAL_ID>
+  <KEY>epic_version</KEY>
+  <VALUE>2.0</VALUE>
+  </EXTENSION>
+  <EXTENSION>
+  <LOCAL_ID>3</LOCAL_ID>
+  <KEY>handle_version</KEY>
+  <VALUE>7.2.3</VALUE>
+  </EXTENSION>
+  </EXTENSIONS>
+</SERVICE_ENDPOINT>
+
+<SERVICE_ENDPOINT PRIMARY_KEY="34JK">
+  <PRIMARY_KEY>34JK</PRIMARY_KEY>
+  <HOSTNAME>epic.grnet.gr</HOSTNAME>
+  <GOCDB_PORTAL_URL>
+    https://creg.eudat.eu/view_portal/index.php?Page_Type=Service&id=384
+  </GOCDB_PORTAL_URL>
+  <BETA>N</BETA>
+  <SERVICE_TYPE>b2handle.handle.api</SERVICE_TYPE>
+  <HOST_IP>62.217.127.244</HOST_IP>
+  <HOST_IPV6>2001:648:2ffc:121:dcdb:eeff:fed6:ba9e/64</HOST_IPV6>
+  <CORE/>
+  <IN_PRODUCTION>Y</IN_PRODUCTION>
+  <NODE_MONITORED>Y</NODE_MONITORED>
+  <SITENAME>GRNET</SITENAME>
+  <COUNTRY_NAME>Greece</COUNTRY_NAME>
+  <COUNTRY_CODE>GR</COUNTRY_CODE>
+  <ROC_NAME>EUDAT_REGISTRY</ROC_NAME>
+  <URL>https://epic.grnet.gr/api/v2/handles/11500</URL>
+  <ENDPOINTS/>
+  <EXTENSIONS/>
+</SERVICE_ENDPOINT>
+
+<SERVICE_ENDPOINT PRIMARY_KEY="45V3">
+  <PRIMARY_KEY>45V3</PRIMARY_KEY>
+  <HOSTNAME>epic.grnet.gr</HOSTNAME>
+  <GOCDB_PORTAL_URL>
+    https://creg.eudat.eu/view_portal/index.php?Page_Type=Service&id=384
+  </GOCDB_PORTAL_URL>
+  <BETA>N</BETA>
+  <SERVICE_TYPE>b2handle.handle.resolver</SERVICE_TYPE>
+  <HOST_IP>62.217.127.244</HOST_IP>
+  <CORE/>
+  <IN_PRODUCTION>Y</IN_PRODUCTION>
+  <NODE_MONITORED>Y</NODE_MONITORED>
+  <SITENAME>GRNET</SITENAME>
+  <COUNTRY_NAME>Greece</COUNTRY_NAME>
+  <COUNTRY_CODE>GR</COUNTRY_CODE>
+  <ROC_NAME>EUDAT_REGISTRY</ROC_NAME>
+  <URL>https://epic.grnet.gr</URL>
+  <ENDPOINTS/>
+  <EXTENSIONS/>
+</SERVICE_ENDPOINT>
+
+<SERVICE_ENDPOINT PRIMARY_KEY="56GL">
+  <PRIMARY_KEY>56GL</PRIMARY_KEY> 
+  <GOCDB_PORTAL_URL>
+    https://creg.eudat.eu/view_portal/index.php?Page_Type=Service&id=5
+  </GOCDB_PORTAL_URL>
+  <BETA>N</BETA>
+  <SERVICE_TYPE>b2handle.handle.api</SERVICE_TYPE>
+  <HOST_IP>130.183.206.85</HOST_IP>
+  <CORE/>
+  <IN_PRODUCTION>Y</IN_PRODUCTION>
+  <NODE_MONITORED>Y</NODE_MONITORED>
+  <SITENAME>MPCDF</SITENAME>
+  <COUNTRY_NAME>Germany</COUNTRY_NAME>
+  <COUNTRY_CODE>DE</COUNTRY_CODE>
+  <ROC_NAME>EUDAT_REGISTRY</ROC_NAME>
+  <URL>https://epic.rzg.mpg.de/epic_v2/34155</URL>
+  <ENDPOINTS/>
+  <EXTENSIONS/>
+</SERVICE_ENDPOINT>
+```
+
+## NCG
+
+ARGO NCG is the connector of the ARGO Monitor Engine for the GOCDB. ARGO NCG expects to retrieve from the GOCDB Hostname, Service Endpoint tuples, which are unique. The following example is a representation of the structure we  we are going to use to store the information we gather from the GOCDB for each service endpoint.
+
+### Below is a perl hash that represents the proposed data model:
+
+```
+'epic.grnet.gr' => {
+  'HOSTNAME' => 'epic.grnet.gr',
+  'SERVICES' => {
+    'b2handle.handle.api' => {
+      'ID' => {
+        '25F8' => {
+          'ATTRIBUTES' => {
+            'b2handle.handle.api_URL' => {
+              'VALUE' => 'https://epic.grnet.gr/api/v2/handles/11239'
```

Comment:
What is the rational behind this scheme? In my point of view it should be either:

```
'b2handle.handle.api_URL' =>  'https://epic.grnet.gr/api/v2/handles/11239'